nucypher/blockchain/eth/interfaces.py

```diff
@@ -508,7 +512,9 @@ def build_contract_transaction(self,
                 # explicitly estimate gas here with block identifier 'latest' if not otherwise specified
                 # as a pending transaction can cause gas estimation to fail, notably in case of worklock refunds.
                 payload['gas'] = contract_function.estimateGas(payload, block_identifier='latest')
+                payload['_is_replacement_tx'] = use_pending_nonce
             transaction_dict = contract_function.buildTransaction(payload)
+            transaction_dict.pop('_is_replacement_tx', None)
```

Comment:
@derekpierre Thanks for your comment. I developed a wrapper that respects existing gas strategies and max gas prices. Unfortunately, it requires this rather ugly workaround. WDYT? Would that solve gas issues to an acceptable degree?

Reply:
@piotr-roslaniec did you mean to instead tag @cygnusv ? 🙂